### PR TITLE
Bluetooth: host: Fix periodic advertising delete

### PR DESCRIPTION
**Pull request by cvinayak**
Fix periodic advertising delete implementation, to backup
the callback structure before deleting the instance so
that the application's terminate callback is called
thereafter.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -4684,10 +4685,11 @@ static void le_per_adv_sync_lost(struct net_buf *buf)
 	/* Deleting before callback, so the caller will be able to restart
 	 * sync in the callback
 	 */
+	cb = per_adv_sync->cb;
 	per_adv_sync_delete(per_adv_sync);
 
-	if (per_adv_sync->cb && per_adv_sync->cb->term) {
-		per_adv_sync->cb->term(per_adv_sync, &term_info);
+	if (cb && cb->term) {
+		cb->term(per_adv_sync, &term_info);
```
Comment (Thalley) [inline subsys/bluetooth/host/hci_core.c:4692]: I think it would be cleaner to just no clean up the callback pointer in the PA sync object, as that doesn't really do anything functionality-wise. Once the flags have been cleared in per_adv_sync_delete, the object will be reused for future `new` PA objects, and the callbacks will always be overwritten, so there's actually no need to clear the callback pointer on delete, aside from it being symbolic. 

I would suggest just to remove the line that clears the callback in `per_adv_sync_delete`.

**Review by Thalley — APPROVED** [review]
LGTM

File: subsys/bluetooth/host/hci_core.c [hunk 2]
```diff
@@ -4667,8 +4666,8 @@ static void le_per_adv_sync_lost(struct net_buf *buf)
 {
 	struct bt_hci_evt_le_per_adv_sync_lost *evt =
 		(struct bt_hci_evt_le_per_adv_sync_lost *)buf->data;
-	struct bt_le_per_adv_sync *per_adv_sync;
 	struct bt_le_per_adv_sync_term_info term_info;
+	struct bt_le_per_adv_sync *per_adv_sync;
```
Comment (jhedberg) [inline subsys/bluetooth/host/hci_core.c:4670]: How is this change related? If it's cleanup it should be a separate commit (can still be same PR)

Comment (cvinayak) [reply subsys/bluetooth/host/hci_core.c:4670]: Fixed.
